src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -146,9 +146,10 @@ private function parseDefinitions($content, $file)
         if (!is_array($content['services'])) {
             throw new InvalidArgumentException(sprintf('The "services" key should contain an array in %s. Check your YAML syntax.', $file));
         }
+        $defaults = isset($content['service_defaults']) ? $content['service_defaults'] : array();
```

Comment:
I think unsupported keys should throw an exception to be consistent with the rest of the loader.